opensearch/examples/index_template.rs

```diff
@@ -0,0 +1,223 @@
+use opensearch::cluster::{ClusterDeleteComponentTemplateParts, ClusterPutComponentTemplateParts};
+use opensearch::indices::{
+    IndicesCreateParts, IndicesDeleteIndexTemplateParts, IndicesDeleteParts,
+    IndicesGetIndexTemplateParts, IndicesGetParts, IndicesPutIndexTemplateParts,
+};
+use opensearch::{
+    cert::CertificateValidation, http::transport::SingleNodeConnectionPool,
+    http::transport::TransportBuilder, OpenSearch,
+};
+use serde_json::json;
+use url::Url;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Let's create a client instance, and an index named `movies`:
+
+    // Create a client to make API calls to OpenSearch running on http://localhost:9200.
+
+    let url = Url::parse("https://localhost:9200")?; 
+    let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
+        .cert_validation(CertificateValidation::None)
+        .build()?;
+    let client = OpenSearch::new(transport);
```

Comment:
Let's stay consistent with https://github.com/opensearch-project/opensearch-rs/pull/146 and have the same setup as that, also the first comment refers to creating an index named movies which we are not doing, so remove it.

```suggestion
    // Create a client to make API calls to OpenSearch running on https://localhost:9200.
    
    let url = Url::parse("https://localhost:9200")?;
    let credentials = Credentials::Basic("admin".into(), "admin".into());
    let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
        .cert_validation(CertificateValidation::None)
        .auth(credentials)
        .build()?;
    let client = OpenSearch::new(transport);
```